ISISScientificComputing/autoreduce
# 482 plot factory

### Summary of work
Implementation of a plot factory to produce Plotly DashApp objects for direct insertion into Django webapp as part of the visualisation project.

### How to test your work
* Check README.md for grammatical errors.
* Check that a DashApp object can be called from factory and used to display a plot locally on 127.0.0.1 (Contact me and I will send over a small script to do this)
* Ensure updating .yaml mode and re-running script changes plot layout e.g. (lines, markers, lines+markers, left blank)
* Check that leaving plot value empty in .yaml will plot a scatter plot by default.
* Ensure all unit tests pass.
* Ensure there is no drop in coverage. 

### Additional comments
When testing if DashApp object can be viewed locally on 127.0.0.1, double click on traces and select only traces of interest. (The DashApp will show all traces by default otherwise which can be overwhelming)

Fixes #482
